tests/pytorch/fpga/test_im2col_conv2d_fpga.py

```diff
@@ -89,7 +101,20 @@ def TransformToFPGA(dace_module):
             vec_type = dace.vector(dace.float32, vec_width)
             utils.vectorize_array_and_memlet(sdfg, "fpga_ONNX_3", vec_type)
 
-        sdfg.expand_library_nodes()
+            # vectorize input too for tiled implementation
+            if "tile" in expansion:
+                utils.vectorize_array_and_memlet(sdfg, "fpga_input", vec_type)
+
+        # pass expansion parameters for tiled implementation
+        if "tile" in expansion:
+            node, state = get_library_node_by_name(sdfg, "Conv_0")
+            node.expand(sdfg,
+                        state,
+                        tiles=(torch_output.shape[3] * 8),
```

Comment:
could you a test with some tile size that is not a multiple of the output shape?

Reply:
To prevent further indexing translations and additional boundary computations in the buffering scheme I only allowed tile sizes which are a multiple of the output image width. I thought this is flexibel enough to allow for a resource - performance tradeoff, while making the implementation significantly simpler.

I can add a test with a different multiple if you like, or also using different number of PEs. PEs are supported in arbitrary numbers.

Reply:
Ok makes sense!